core/extractors/extractors_docling.py

```
@@ -0,0 +1,430 @@
+        """
+        pdf_path = Path(pdf_path)
+        
+        if not pdf_path.exists():
+            raise FileNotFoundError(f"PDF not found: {pdf_path}")
+        
+        # Pre-validation: Check if file is readable and not empty
+        file_size = pdf_path.stat().st_size
+        if file_size == 0:
+            # More specific error message based on file type
+            if pdf_path.suffix in ['.txt', '.text', '.md']:
+                raise RuntimeError(f"Text file is empty: {pdf_path}")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)